staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go

```diff
@@ -295,10 +320,13 @@ func (c *Controller) syncPool(ctx context.Context, poolName string) error {
 	logger := klog.FromContext(ctx)
 
 	// Gather information about the actual and desired state.
-	// TODO: index by pool name.
 	var slices []*resourceapi.ResourceSlice
-	for _, obj := range c.sliceStore.List() {
-		if slice, ok := obj.(*resourceapi.ResourceSlice); ok && slice.Spec.Pool.Name == poolName {
+	objs, err := c.sliceStore.ByIndex(poolNameIndex, poolName)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve ResourceSlice objects: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("retrieve ResourceSlice objects: %w", err)
```

Reply:
I should check other places and make changes. I did not do it well.